### PR TITLE
Update vendor & fix breaking scheduler changes

### PR DESCRIPTION
**Pull request by towca**
The following things changed in scheduler and needed to be fixed:
* NodeInfo was moved to schedulerframework
* Some fields on NodeInfo are now exposed directly instead of via getters
* NodeInfo.Pods is now a list of *schedulerframework.PodInfo, not *apiv1.Pod
* SharedLister and NodeInfoLister were moved to schedulerframework
* PodLister was removed

### CODE REVIEW

File: cluster-autoscaler/simulator/delta_cluster_snapshot.go [hunk 1]
```diff
@@ -51,24 +49,24 @@ type deltaSnapshotPodLister DeltaClusterSnapshot
...
-	modifiedNodeInfoMap map[string]*schedulernodeinfo.NodeInfo
+	addedNodeInfoMap    map[string]*schedulerframework.NodeInfo
+	modifiedNodeInfoMap map[string]*schedulerframework.NodeInfo
 	deletedNodeInfos    map[string]bool
 
-	nodeInfoList         []*schedulernodeinfo.NodeInfo
+	nodeInfoList         []*schedulerframework.NodeInfo
 	podList              []*apiv1.Pod
-	havePodsWithAffinity []*schedulernodeinfo.NodeInfo
+	havePodsWithAffinity []*schedulerframework.NodeInfo
```
Comment (MaciekPytel) [inline cluster-autoscaler/simulator/delta_cluster_snapshot.go:58]: Is podList still used for anything after removing the pod lister functionality?

Comment (towca) [reply cluster-autoscaler/simulator/delta_cluster_snapshot.go:56]: No, good catch - fixed.

File: cluster-autoscaler/simulator/delta_cluster_snapshot.go [hunk 2]
```diff
@@ -168,20 +164,11 @@ func (data *internalDeltaSnapshotData) addNodeInfo(nodeInfo *schedulernodeinfo.N
 		data.nodeInfoList = append(data.nodeInfoList, nodeInfo)
 	}
 
-	if len(nodeInfo.Pods()) > 0 {
-		data.clearPodCaches()
-	}
```
Comment (MaciekPytel) [inline cluster-autoscaler/simulator/delta_cluster_snapshot.go:173]: I think we still need to drop havePodsWithAffinity here.

Comment (MaciekPytel) [reply cluster-autoscaler/simulator/delta_cluster_snapshot.go:173]: cc: @aleksandra-malinowska

Comment (towca) [reply cluster-autoscaler/simulator/delta_cluster_snapshot.go:173]: Ohh right, that's my bad!

**Comment by MaciekPytel** [comment]
lgtm modulo the cache invalidation comment. Thanks for doing this!

Also a minor general comment for future PRs - if trivial import changes happened in separate comment it would be easier to focus review on non-trivial parts.

**Pull request merged**